lib/sycamore/sycamore/transforms/detr_partitioner.py

```diff
@@ -554,7 +552,25 @@ def _partition_pdf_batched_named(
             with LogTime("pdfminer_supplement"):
                 for d, p in zip(deformable_layout, pdfminer_layout):
                     self._supplement_text(d, p)
+        for i in batches:
```

Comment:
also suggest adding a test to catch the pdfminer dependency for extraction (it looks like that was the problem), to make sure it doesn't regress.

Reply:
Ok, I just took another look. This shouldn't be doing anything. batches is a generator, but it was consumed in the first pass.  If it's not, then we have a memory issue since you'd be keeping the entire pdf images in memory.

```
def generator(n):
    for i in range(n):
        print(f"yielding {n}")
        yield i

f = generator(10)
for i in f:
    print(f"got {i}")

print("LOOP 2")
for i in f:
    print(f"got {i}")
```

Reply:
Fixed, thanks for catching this

Reply:
Fixed